crypto/core_namemap.c

```diff
@@ -173,6 +173,32 @@ int ossl_namemap_name2num(const OSSL_NAMEMAP *namemap, const char *name)
     return number;
 }
 
+struct num2name_data_st {
+    size_t idx;                  /* Countdown */
+    const char *name;            /* Result */
+};
+
+static void do_num2name(const char *name, void *vdata)
+{
+    struct num2name_data_st *data = vdata;
+
+    if (data->idx > 0)
+        data->idx--;
+    else if (data->name == NULL)
+        data->name = name;
+}
+
+const char *ossl_namemap_num2name(const OSSL_NAMEMAP *namemap, int number,
```

Comment:
Currently it seems this function is only ever called once internally, and the idx parameter is always 0

Reply:
*Currently*, yes.  But then, we only have one name per method for the moment being.  I did imagine that some might want to iterate over all names using an index...  on the other hand, I'm planning on `doall_name` functionality (I've a branch for it already), so it might not be necessary...  I'll have a think on this.

Reply:
I think I will keep it like this for now.